fe/fe-core/src/main/java/com/starrocks/authentication/LDAPAuthProviderForExternal.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.authentication;
+
+import com.starrocks.mysql.privilege.Password;
+import com.starrocks.sql.ast.UserIdentity;
+import org.apache.commons.lang3.StringUtils;
+
+import static com.starrocks.mysql.privilege.AuthPlugin.AUTHENTICATION_LDAP_SIMPLE_FOR_EXTERNAL;
+
+public class LDAPAuthProviderForExternal implements AuthenticationProvider {
+    public static final String PLUGIN_NAME = AUTHENTICATION_LDAP_SIMPLE_FOR_EXTERNAL.name();
+
+    @Override
+    public UserAuthenticationInfo validAuthenticationInfo(UserIdentity userIdentity,
+                                                          String password, String textForAuthPlugin)
+            throws AuthenticationException {
+        throw new AuthenticationException("unsupported");
+    }
+
+    public static boolean authenticate(String username, String password,
+                                       String host, String port,
+                                       String baseDn, String searchAttr,
+                                       String rootDn, String rootPwd) throws Exception {
+        return true;
```

Comment:
the sonarcloud failure can be ignore for now, because we extract some logic here.